### PR TITLE
closes #3354: Update v7 migration guide

### LINKED ISSUE #3354: Update `v7` migration guide for removing `personal` namespace
#2844 removed support for the `personal` namespace. We should document in the `v7` migration guide that unlocking / locking accounts is no longer supported along with the `personal` namespace removal. A link to the go-ethereum docs for using *clef* and a mention to providing a keyfile as alternatives might be good notes to add as well.

### PR DESCRIPTION
**Pull request by fselmo**
Closes # #3354

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
lgtm

File: docs/v7_migration.rst [hunk 1]
```diff
@@ -224,6 +208,42 @@ errors:
...
+Python 3.7 support has been dropped in favor of Python 3.8+. Python 3.7 is no longer
+supported by the Python core team, and we want to focus our efforts on supporting
+the latest versions of Python.
+
+
+EthPM Module Removed
+````````````````````
+
+The EthPM module has been removed from the library. It was not widely used and has not
+been functional since around October 2022. It was deprecated in ``v6`` and has been
```
Comment (kclowes) [inline docs/v7_migration.rst:226]: just a passing curiosity - it hasn't been functional? or the tests haven't been functional?

File: docs/v7_migration.rst [hunk 2]
```diff
@@ -224,6 +208,42 @@ errors:
...
+completely removed in ``v7``.
+
+
+Geth Miner Namespace Removed
+````````````````````````````
+
+The ``geth.miner`` namespace, deprecated in ``v6``, has been removed in ``v7``.
```
Comment (kclowes) [inline docs/v7_migration.rst:233]: I may be looking at the wrong thing, but I don't think we deprecated miner and personal? I think we may have just removed them?

**Review by kclowes — APPROVED** [review]
I left a couple nits, but looks good to me! Has anyone gone through and audited to make sure that everything that's in v7 got added here?

File: docs/v7_migration.rst (hunk 1 above)
Comment (fselmo) [reply docs/v7_migration.rst:226]: Not sure tbh. I just moved this section over. This was reviewed and merged somewhere else.

File: docs/v7_migration.rst (hunk 2 above)
Comment (fselmo) [reply docs/v7_migration.rst:233]: Welp, I s'pose those should be backported 😅

File: docs/v7_migration.rst (hunk 1 above)
Comment (pacrob) [reply docs/v7_migration.rst:226]: Using it relied on a free IPFS portal provided by Infura. Once Infura started requiring an API key to access it (in October '22), EthPM no longer worked. I looked at updating it to allow the user to provide their own key, but we determined that it wasn't getting enough use to make the updates worthwhile.

Comment (kclowes) [reply docs/v7_migration.rst:226]: got it, thanks! 🚢

Comment (fselmo) [reply docs/v7_migration.rst:226]: > Using it relied on a free IPFS portal provided by Infura. Once Infura started requiring an API key to access it (in October '22), EthPM no longer worked. I looked at updating it to allow the user to provide their own key, but we determined that it wasn't getting enough use to make the updates worthwhile.

Ah yeah, now I remember that context 👌🏼

**Pull request merged**